examples/model_selection/plot_roc.py

```diff
@@ -1,132 +1,269 @@
+under the curve (AUC) is usually better.
 
 The "steepness" of ROC curves is also important, since it is ideal to maximize
 the true positive rate while minimizing the false positive rate.
 
-ROC curves are typically used in binary classification to study the output of
-a classifier. In order to extend ROC curve and ROC area to multi-label
-classification, it is necessary to binarize the output. One ROC
-curve can be drawn per label, but one can also draw a ROC curve by considering
-each element of the label indicator matrix as a binary prediction
-(micro-averaging).
+ROC curves are typically used in binary classification, where the TPR and FPR
+can be defined unambiguously. In the case of multiclass classification, a notion
+of TPR/FPR is obtained only after binarizing the output. This can be done in 2
+different ways:
 
-Another evaluation measure for multi-label classification is
-macro-averaging, which gives equal weight to the classification of each
-label.
+- the One-vs-Rest scheme compares each class against all the others (assumed as
+  one);
+- the One-vs-One scheme compares every unique pairwise combination of classes.
 
-.. note::
+In this example we explore both schemes and demo the concepts of micro and macro
+averaging as different ways of summarizing the information of the multiclass ROC
+curves.
 
-    See also :func:`sklearn.metrics.roc_auc_score`,
-             :ref:`sphx_glr_auto_examples_model_selection_plot_roc_crossval.py`
+.. note::
 
+    See :ref:`sphx_glr_auto_examples_model_selection_plot_roc_crossval.py` for
+    an extension of the present example estimating the variance of the ROC
+    curves and their respective AUC.
 """
 
-import numpy as np
-import matplotlib.pyplot as plt
-from itertools import cycle
+# %%
+# Load and prepare data
+# =====================
+#
+# We import the :ref:`iris_dataset` which contains 3 classes, each one
+# corresponding to a type of iris plant. One class is linearly separable from
+# the other 2; the latter are **not** linearly separable from each other.
+#
+# Here we binarize the output and add noisy features to make the problem harder.
 
-from sklearn import svm, datasets
-from sklearn.metrics import roc_curve, auc
+import numpy as np
+from sklearn.datasets import load_iris
 from sklearn.model_selection import train_test_split
-from sklearn.preprocessing import label_binarize
-from sklearn.multiclass import OneVsRestClassifier
-from sklearn.metrics import roc_auc_score
-
-# Import some data to play with
-iris = datasets.load_iris()
-X = iris.data
-y = iris.target
 
-# Binarize the output
-y = label_binarize(y, classes=[0, 1, 2])
-n_classes = y.shape[1]
+iris = load_iris()
+target_names = iris.target_names
+X, y = iris.data, iris.target
+y = iris.target_names[y]
 
-# Add noisy features to make the problem harder
 random_state = np.random.RandomState(0)
 n_samples, n_features = X.shape
+n_classes = len(np.unique(y))
 X = np.c_[X, random_state.randn(n_samples, 200 * n_features)]
```

Comment:
I don't like this `np.c_` :). Can we use `np.concatenate` instead. It much more friendly.